tests/unit/test_coordinate_utilities.py

```diff
@@ -543,3 +520,209 @@ def test_any_absent_dimension_variables(self):
+                    [89.3, -9999, 60.3, 59.3, 29.3, 2.1, -9999, -9999, -9999, -89.3],
+                    [
+                        -9999,
+                        79.3,
+                        -60.3,
+                        -9999,
+                        -9999,
+                        -9999,
+                        -60.2,
+                        -59.3,
+                        -79.3,
+                        -89.3,
+                    ],
+                    [
+                        -89.3,
+                        79.3,
+                        -60.3,
+                        -9999,
+                        -9999,
+                        -9999,
+                        -60.2,
+                        -59.3,
+                        -79.3,
+                        -9999,
+                    ],
+                ]
+            )
+            expected_geo_grid_points_reversed = (
+                [(-179.3, 89.3), (-179.3, -89.3)],
+                [(-179.3, 89.3), (178.4, -89.3)],
+            )
+            actual_geo_grid_points_reversed = get_row_col_geo_grid_points(
+                lat_arr_reversed,
+                lon_arr_reversed,
+                self.varinfo.get_variable(self.latitude),
+                self.varinfo.get_variable(self.longitude),
+            )
+
+            self.assertListEqual(
+                actual_geo_grid_points_reversed[0], expected_geo_grid_points_reversed[0]
+            )
+            self.assertListEqual(
+                actual_geo_grid_points_reversed[1], expected_geo_grid_points_reversed[1]
+            )
+
+    def test_get_max_x_spread_pts(self):
+        """Ensure that two valid sets of indices are returned by the function
+        with a masked dataset as input
+
+        """
+
+        with self.subTest('Get two sets of valid indices for points from coordinates'):
+            valid_values = np.array(
+                [
+                    [True, True, True, True, False, False, True, True, True, True],
+                    [True, True, True, False, False, False, True, True, True, True],
+                    [True, True, True, False, True, False, True, True, True, True],
+                    [True, True, True, False, False, False, True, True, True, True],
+                    [True, True, False, False, False, False, True, True, True, True],
+                ]
+            )
+            expected_indices = [[0, 0], [0, 9]]
+
+            actual_indices = get_max_x_spread_pts(~valid_values)
+
+            self.assertTrue(actual_indices[0] == expected_indices[0])
+            self.assertTrue(actual_indices[1] == expected_indices[1])
+
+    def test_get_row_col_valid_indices_in_dataset(self):
+        """Ensure that two sets of valid indices are
+        returned by the method with a set of lat/lon coordinates as input
+
+        """
+        with self.subTest('Get two sets of valid indices from coordinates dataset'):
```

Comment:
Similar comments to a couple of other places. This unit test is only really testing the happy-path. You aren't really throwing any potential edge-cases at the function. Other test cases are:

* Only a single valid pixel.
* Only valid pixels in a single row.
* Only valid pixels in a single column.
* No valid pixels.

These seem like overkill, but it's really important to know how this code behaves in realistic edge-cases, so we know failure is graceful.

Reply:
updated unit tests - https://github.com/nasa/harmony-opendap-subsetter/pull/18/commits/57744a6298adf081d0bf06c9a6acd0b98cccf99c

Reply:
Nice - thanks!